### PR TITLE
Do not block distribution when one fails

### PR DESCRIPTION
**Pull request by hschallhorn**
References #12932

### Description
Moves transaction block to a single appeal rather than a whole distribution. This ensures `JudgeAssignTask` creation, `DistributionTask` closure, and `DistributedCase` creation is done inside of a transaction, so if one fails, the appeal remains unchanged, but the distribution can continue. 

### Acceptance Criteria
- [ ] Case distribution does not fail in the event that a single DistributedCase instantiation fails.
- [ ] Development team is reliably informed of the failure, so that the appeal can be fixed and the case can proceed.

### What this does not solve
 - Ensure that the total number of requested distributions are created

### CODE REVIEW

File: spec/models/dockets/hearing_request_docket_spec.rb [hunk 1]
```diff
@@ -203,6 +203,35 @@
...
+
+    context "when an appeal aleady has a distribution" do
+      subject do
+        HearingRequestDocket.new.distribute_appeals(distribution, priority: false, limit: 10, genpop: "any")
+      end
+
+      it "does not fail and distributes the legitimate tasks" do
+        number_of_already_distributed_appeals = 1
+        total_number_of_appeals = 10
+        total_number_of_appeals.times { create_nonpriority_distributable_hearing_appeal_not_tied_to_any_judge }
```
Comment (hschallhorn) [inline spec/models/dockets/hearing_request_docket_spec.rb]: Create a bunch of appeals to distribute

File: app/workflows/hearing_request_case_distributor.rb [hunk 2]
```diff
@@ -9,39 +11,40 @@ def initialize(appeals:, genpop:, distribution:, priority:)
...
+    not_genpop_appeals.map { |appeal| [appeal, false] }.concat(only_genpop_appeals.map { |appeal| [appeal, true] })
   end
 
-  def assign_judge_tasks_for_not_genpop_appeals
-    CreateJudgeAssignTasksForAppeals.new(appeals: not_genpop_appeals, judge: distribution.judge).call.map do |task|
-      [task, false]
-    end
+  def create_judge_assign_task_for_appeal(appeal)
+    JudgeAssignTaskCreator.new(appeal: appeal, judge: distribution.judge).call
   end
```
Comment (hschallhorn) [inline app/workflows/hearing_request_case_distributor.rb]: Essentially what [CreateJudgeAssignTasksForAppeals](https://github.com/department-of-veterans-affairs/caseflow/blob/c08b7acc77fcdf50d9947e6b6f80bae005bd3a02/app/workflows/create_judge_assign_tasks_for_appeals.rb#L13) does anyway.

File: spec/models/dockets/hearing_request_docket_spec.rb [hunk 3]
```diff
@@ -203,6 +203,42 @@
...
```
Comment (hschallhorn) [inline spec/models/dockets/hearing_request_docket_spec.rb]: Create some appeals to distribute

File: spec/models/dockets/hearing_request_docket_spec.rb [hunk 4]
```diff
@@ -203,6 +203,42 @@
...
+        create(:staff, :judge_role, sdomainid: previous_distribution_judge.css_id)
+        previous_distribution = Distribution.create!(judge: previous_distribution_judge)
+        HearingRequestDocket.new.distribute_appeals(previous_distribution,
+                                                    priority: false,
+                                                    limit: number_of_already_distributed_appeals,
+                                                    genpop: "any")
+        distributed_appeals = DistributionTask.closed.take(number_of_already_distributed_appeals).map(&:appeal)
+        distributed_appeals.each do |distributed_appeal|
+          DistributionTask.create!(appeal: distributed_appeal, parent: distributed_appeal.root_task)
+        end
```
Comment (hschallhorn) [inline spec/models/dockets/hearing_request_docket_spec.rb]: Distribute some of the appeals and make them ready to distribute again

File: app/workflows/hearing_request_case_distributor.rb [hunk 5]
```diff
@@ -9,39 +11,40 @@ def initialize(appeals:, genpop:, distribution:, priority:)
...
 
-  def assign_judge_tasks_for_only_genpop_appeals
-    CreateJudgeAssignTasksForAppeals.new(appeals: only_genpop_appeals, judge: distribution.judge).call.map do |task|
-      [task, true]
-    end
+  def create_distribution_case_for_task(task, genpop_value)
+    distribution.distributed_cases.create!(
+      case_id: task.appeal.uuid,
+      docket: "hearing",
```
Comment (kevmo) [inline app/workflows/hearing_request_case_distributor.rb]: Nitpick: Seems like this should be a constant.

**Pull request merged**